src/main/java/gov/hhs/cms/bluebutton/fhirstress/backend/CustomSamplerClient.java

```diff
@@ -0,0 +1,143 @@
+	 */
+	@Override
+	public Arguments getDefaultParameters() {
+		Arguments defaultParameters = new Arguments();
+		defaultParameters.addArgument(PARAM_SERVER, "https://fhir.backend.bluebutton.hhsdevcloud.us/baseDstu3");
+		defaultParameters.addArgument(KEYSTORE_DIR, "/opt/fhir_stress/dev/ssl-stores");
+		defaultParameters.addArgument(PROXY_HOST, "null");
+		defaultParameters.addArgument(PROXY_PORT, "0");
+		defaultParameters.addArgument(RIFFILE, "beneficiary_test.rif");
+		defaultParameters.addArgument(DELIMITER, "|");
+		return defaultParameters;
+	}
+
+	/**
+	 * @see org.apache.jmeter.protocol.java.sampler.AbstractJavaSamplerClient#setupTest(org.apache.jmeter.protocol.java.sampler.JavaSamplerContext)
+	 */
+	@Override
+	public void setupTest(JavaSamplerContext context) {
+		super.setupTest(context);
+		this.hostName = getHostname();
+    int proxyPort = Integer.parseInt(context.getParameter(PROXY_PORT));
+
+    if(context.getParameter(PROXY_HOST).equals("null")) { // no proxy
+		  this.client = FhirClient.create(
+        context.getParameter(PARAM_SERVER),
+        context.getParameter(KEYSTORE_DIR)
+      );
+    }
+    else { // use proxy
+		  this.client = FhirClient.create(
+        context.getParameter(PARAM_SERVER),
+        context.getParameter(KEYSTORE_DIR),
+        context.getParameter(PROXY_HOST),
+        Integer.parseInt(context.getParameter(PROXY_PORT))
+      );
+    }
+    //this.parser = new RifParser(context.getParameter(RIFFILE), context.getParameter(DELIMITER));
+    //System.out.println("Thread loops = " + context.getParameter(LOOPS));
+
+		// Find the IDs that can be queried in each sample run.
+		//this.patientIds = findPatientIds();
+	}
+
+	/**
+	 * @see org.apache.jmeter.protocol.java.sampler.JavaSamplerClient#runTest(org.apache.jmeter.protocol.java.sampler.JavaSamplerContext)
+	 */
+	@Override
+	public SampleResult runTest(JavaSamplerContext context) {
+		SampleResult sample = new SampleResult();
+		sample.sampleStart();
+		try {
+			executeTest();
+
+			sample.sampleEnd();
+			sample.setSuccessful(true);
+			sample.setResponseMessage("Sample succeeded on host: " + hostName);
+			sample.setResponseCodeOK();
+		} catch (IllegalArgumentException|InvalidRequestException|FhirClientConnectionException e) {
+			// Mark this sample iteration as failed.
+			sample.sampleEnd();
+			sample.setSuccessful(false);
+			sample.setResponseMessage("Exception: " + e);
+
+			/*
+			 * Serialize the stack trace to a String and attach it to the sample
+			 * results.
+			 */
+			StringWriter stringWriter = new StringWriter();
+			e.printStackTrace(new java.io.PrintWriter(stringWriter));
+			sample.setResponseData(stringWriter.toString(), StandardCharsets.UTF_8.name());
+			sample.setDataType(org.apache.jmeter.samplers.SampleResult.TEXT);
+      if (e instanceof FhirClientConnectionException) {
+			  sample.setResponseCode("504");
+      }
+      else {
+			  sample.setResponseCode("500");
+      }
+		}
```

Comment:
Not a big fan of the error handling here: you're assuming that an abstract method you're calling from a subclass will only throw those exact exceptions. I'd either make it robust enough to handle any exception, or move the error handling down into the subclasses. (I'd recommend the latter.)

Reply:
This is mapping an exception into a jmeter response code that I can see in the log.  If I recall, the FHIR client was having connection issues at the time but it was really hard to find in the jmeter log.  So this just maps it into a 504 which is like "gateway timeout".  All other exceptions get the default 500 moniker which I believe what was previously happening(it onlyd handled IllegalArgumentException and InvalidRequestException originally).  I think without the handler for FhirClientConnectionException here you get no visibility if something goes awry.  We could add Exception to the catch but not sure it is completely necessary at this time.

Reply:
:+1: